director/redirect.go

```diff
@@ -348,9 +355,56 @@ func RegisterOrigin(ctx *gin.Context) {
 	ctx.JSON(200, gin.H{"msg": "Successful registration"})
 }
 
+// Return a list of available origins URL in Prometheus HTTP SD format
+// for director's Prometheus service discovery
+func DiscoverOrigins(ctx *gin.Context) {
+	// Check token for authorization
+	tokens, present := ctx.Request.Header["Authorization"]
+	if !present || len(tokens) == 0 {
+		ctx.JSON(401, gin.H{"error": "Bearer token not present in the 'Authorization' header"})
+		return
+	}
+	token := strings.TrimPrefix(tokens[0], "Bearer ")
+	ok, err := VerifyDirectorSDToken(token)
+	if err != nil {
+		log.Warningln("Failed to verify director service discovery token:", err)
+		ctx.JSON(401, gin.H{"error": fmt.Sprintf("Authorization token verification failed: %v\n", err)})
+		return
+	}
+	if !ok {
+		log.Warningf("Invalid token for accessing director's sevice discovery")
+		ctx.JSON(401, gin.H{"error": "Invalid token for accessing director's sevice discovery"})
+		return
+	}
+
+	serverAdMutex.RLock()
+	defer serverAdMutex.RUnlock()
+	serverAds := serverAds.Keys()
+	promDiscoveryRes := make([]PromDiscoveryItem, 0)
+	for _, ad := range serverAds {
+		// We don't include caches in this discovery for right now
+		if ad.Type != OriginType {
+			continue
+		}
+		promDiscoveryRes = append(promDiscoveryRes, PromDiscoveryItem{
+			// TODO: change to ad.WebURL when #285 is ready
```

Comment:
I believe this PR was merged; let's address this item prior to merging the branch.